Add TypeScript support in apps/

maddiedierker opened the pull request:
Allows for incremental TypeScript adoption with .ts and .tsx files. Changing TypeScript files will also cause the frontend server to rebuild automatically.

The Student Learning team has proposed that we use TypeScript for the "Lab Lab" project (an effort to build a shared platform for new/existing labs). More details about the project can be found in [Lab Lab Tenets](https://docs.google.com/document/d/1M-tHYj7fcfmuVBWZKjVX0hx3-jQNua8OdzlxgFnpCF8/edit).

**Follow-up work:**
- We'll likely need to add some linting rules or extend existing ones to work for TypeScript, but that can come later.

molly-moen reviewed (approved):
🎉

cat5inthecradle commented:
:heart_eyes:

davidsbailey commented:
Very exciting! Do you mind saying what you did to test this? in particular I am curious about:
* `yarn start`
* `yarn test:entry --entry=...`

not sure if these can be tested yet, but would be nice to track somewhere to make sure these development scenarios work

maddiedierker commented:
@davidsbailey 
> Do you mind saying what you did to test this?

after making the commit in this PR, i did some manual testing:
1. added a `.ts` file with a typed function
2. called that function from an existing `.js` file
3. restarted my frontend server (`yarn start`)
4. confirmed that the `.ts` function successfully executed
5. changed the `.ts` file and confirmed the frontend server rebuilt

i didn't do any testing around unit tests -- i'll do that in a separate PR. i think i need to add at least one more package for that (types for jest, possibly enzyme types as well)

davidsbailey commented:
Sounds great @madelynkasula ! I'm excited for Jest too!

maddiedierker commented:
[posting here for posterity] i'm not sure what i did wrong, but there were additional `yarn.lock` changes that weren't committed, which caused the build to fail (because `yarn install` resulted in an uncommitted `yarn.lock` diff). here's the additional commit: https://github.com/code-dot-org/code-dot-org/commit/0848993c66538c96f195f1bd06866fac0657be26

i SSH'd into the staging machine, committed the `yarn.lock` changes, and pushed the commit
